conda_build/environ.py

```diff
@@ -965,7 +966,11 @@ def get_install_actions(
             with capture():
                 try:
                     precs = _install_actions(prefix, index, specs)["LINK"]
-                except (NoPackagesFoundError, UnsatisfiableError) as exc:
+                except (
+                    NoPackagesFoundError,
+                    UnsatisfiableError,
+                    PackagesNotFoundError,
```

Comment:
@kenodegard, why did we had this change in https://github.com/conda/conda-build/commit/7acbbf200be198e978e08391436695db14298a05 ?
Re-adding `PackagesNotFoundError` fixes gh-5256. Am I missing something?

Reply:
I had added it in the prior commit (https://github.com/conda/conda-build/pull/5203/commits/7539a73734644adafca4871934a173d7b345c0cf) as I was trying to debug a weird error (see https://github.com/conda/conda-build/pull/5203/commits/7acbbf200be198e978e08391436695db14298a05#r1517170310)

Since the fix had nothing to do with catching `PackagesNotFoundError` I removed it again

Reply:
Ah, I landed at 7acbbf2 as the first commit to show the regression but didn't realize that was an intermediate change. Thanks, that's very helpful.
Since https://github.com/conda/conda-build/pull/5203/commits/4da3bce81937b551a02df623dba655e7f3afb1f6 works, there must be something in https://github.com/conda/conda-build/pull/5203/commits/ca792ee3a719aad068d918de18ea4bfe527c9c4b (with the `get_install_actions` name dance applied) that causes it, I think. Let me dig into those changes a bit..

Reply:
Oh, I guess `24.1` didn't show this because https://github.com/conda/conda-build/pull/5152/commits/d7a975158eaa40e5c173955cf194ccb105df5bec broke the "`get_install_actions` is in the last N entries on the stack" assumption of `conda-libmamba-solver`.
Meaning, `conda-build<24.1` also didn't handle these errors correctly and in fact `conda-build=24.1` broke `conda-libmamba-solver` setting `context.offline`...

Reply:
Do we need to adjust anything in conda-libmamba-solver?